stable/dotnetcore/src/Kubeless.Core/Handlers/DefaultParameterHandler.cs

```diff
@@ -37,11 +38,12 @@ private async Task<Event> GetEvent(HttpRequest request)
                 request.Body.Position = 0;
             }
 
-            using var sr = new StreamReader(request.Body);
-            object data = await sr.ReadToEndAsync();
-
-            if (contentType == "application/json" && data.ToString().Length > 0) {
-                data = JsonSerializer.Deserialize<dynamic>(data.ToString());
+            object data;
+            if (contentType.Any(ct => ct.Contains("application/json")) && request.ContentLength > 0) {
+                data = await JsonSerializer.DeserializeAsync<dynamic>(request.Body);
+            } else {
+                using var sr = new StreamReader(request.Body, leaveOpen: true);
+                data = await sr.ReadToEndAsync();
             }

```

Comment:
- use builtin DeserializeAsync method of JsonSerializer 
- fix content-type parsing in some cases (e.g. "Content-Type: application/json; charset=utf-8")